app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def update_user_posts_count
+    author.update(posts_count: author.posts.count)
```

Comment:
- [ ] This should be `post_counter`, not `posts_count` based on your schema configuration.